src/state/opamSwitchState.ml

```diff
@@ -1346,7 +1346,7 @@ let dependencies_filter_to_formula_t ~build ~post st nv =
   OpamFilter.filter_formula ~default:true env
 
 let dependencies_t st base_deps_compute deps_compute
-    ~depopts ~installed ?(unavailable=false) packages =
+    ~depopts ~installed ~unavailable packages =
```

Comment:
How this change speeds up dependencies computation?

Reply:
it doesn't, but as the dedicated commit says:
> Make ?unavailable a non-optional argument to enforce speedups when availability information is not needed

it only enforces that the availability information is used purposefully and ensures that regression won't happen, as the default `unavailable=false` makes it so that the availability information is used and thus the command will take significantly longer

Reply:
Yup, I've read the commit message.
The optional argument `unavailable` will anyway be used on purpose as when calling the function if we want unavailable packages we need to specify it and usually we don't want it. It won't change ow the function is used to have the argument optional or just labelled. As changes show, all omitted `unavailable` argument were false, as when we need it, it is already to true.
I'm not convinced, but I won't request a change. It's worth mentioning it in the documentation of the function, that requesting only for available package will add an extra computation time.

Reply:
> As changes show, all omitted `unavailable` argument were false, as when we need it, it is already to true.

I don't think that's an viable argument. We could've changed the behaviour of the commands that use this function if we had known and this holds for future change as well. Also i haven't checked that each commands actually require `~unavailable:false`. I simply used its previous default to avoid breaking the current behaviour and also because it's outside of the scope of this PR.

> It's worth mentioning it in the documentation of the function, that requesting only for available package will add an extra computation time.

I agree. What do you think of https://github.com/ocaml/opam/pull/5317/commits/e5277ca677b388e9ea951a2228621b9577f1ef3f ?

Reply:
> What do you think of https://github.com/ocaml/opam/commit/e5277ca677b388e9ea951a2228621b9577f1ef3f ?

lgtm! A squash with 260a06e2dbb501a094925f8e0b6a33c5eeb1e6a1 and it's good to go!

Reply:
done